src/lib/facebooknotificationsdatabase.cpp

```diff
@@ -332,16 +349,27 @@ bool FacebookNotificationsDatabase::write()
     const QMap<int, QList<FacebookNotification::ConstPtr> > insertNotifications = d->queue.insertNotifications;
     const QList<int> removeNotificationsFromAccounts = d->queue.removeNotificationsFromAccounts;
     QStringList removeNotifications = d->queue.removeNotifications;
+    bool removeAll = d->queue.removeAll;
 
     d->queue.insertNotifications.clear();
     d->queue.removeNotificationsFromAccounts.clear();
     d->queue.removeNotifications.clear();
+    d->queue.removeAll = false;
 
     locker.unlock();
 
     bool success = true;
     QSqlQuery query;
 
+    if (removeAll) {
+        QVariantList accountIds;
+        accountIds.append(-1);
+
+        query = prepare(QStringLiteral("DELETE FROM notifications WHERE accountId > :accountId"));
```

Comment:
That's what I had at first but it gives me "Failed to execute query", the only way I was able to get it work was to specify WHERE condition (I'm running unit tests on Ubuntu desktop, might be sqlite version related).